src/framework/components/light/component.js

```diff
@@ -178,6 +54,841 @@ class LightComponent extends Component {
+    }
+
+    /**
+     * If enabled and bake=true, the light's direction will contribute to
+     * directional lightmaps. Be aware, that directional lightmap is an approximation and can only hold
+     * single direction per pixel. Intersecting multiple lights with bakeDir=true may lead to incorrect
+     * look of specular/bump-mapping in the area of intersection. The error is not always visible
+     * though, and highly scene-dependent.
+     *
+     * @type {boolean}
+     */
+    set bakeDir(arg) {
+        this._setValue('bakeDir', arg, function (newValue, oldValue) {
+            this.light.bakeDir = newValue;
+        });
+    }
+
+    get bakeDir() {
+        return this.data.bakeDir;
+    }
+
+    /**
+     * Mark light as non-movable (optimization).
+     *
+     * @type {boolean}
+     */
+    set isStatic(arg) {
+        this._setValue('isStatic', arg, function (newValue, oldValue) {
+            this.light.isStatic = newValue;
+        });
+    }
+
+    get isStatic() {
+        return this.data.isStatic;
+    }
+
+    /**
+     * An array of layer IDs ({@link Layer#id}) to which this light should
+     * belong. Don't push/pop/splice or modify this array, if you want to change it - set a new one
+     * instead.
+     *
+     * @type {number[]}
+     */
+    set layers(arg) {
+        this._setValue('layers', arg, function (newValue, oldValue) {
+            for (let i = 0; i < oldValue.length; i++) {
+                const layer = this.system.app.scene.layers.getLayerById(oldValue[i]);
+                if (!layer) continue;
+                layer.removeLight(this);
+                this.light.removeLayer(layer);
+            }
+            for (let i = 0; i < newValue.length; i++) {
+                const layer = this.system.app.scene.layers.getLayerById(newValue[i]);
+                if (!layer) continue;
+                if (this.enabled && this.entity.enabled) {
+                    layer.addLight(this);
+                    this.light.addLayer(layer);
+                }
+            }
+        });
+    }
+
+    get layers() {
+        return this.data.layers;
+    }
+
+    /**
+     * Returns an array of SHADOWUPDATE_ settings per shadow cascade, or undefined if not used.
+     *
+     * @type {number[] | null}
+     */
+    set shadowUpdateOverrides(values) {
+        this.light.shadowUpdateOverrides = values;
+    }
+
+    get shadowUpdateOverrides() {
+        return this.light.shadowUpdateOverrides;
+    }
+
+    /**
+     * Size of penumbra for contact hardening shadows. For area lights
+     * acts as a multiplier with the dimensions of the area light. For punctual and directional lights
+     * it's the area size of the light. Defaults to 1.0.
+     *
+     * @type {number}
+     */
+    set penumbraSize(value) {
+        this.light.penumbraSize = value;
+    }
+
+    get penumbraSize() {
+        return this.light.penumbraSize;
+    }
+
+    /** @ignore */
+    _setValue(name, value, setFunc, skipEqualsCheck) {
```

Comment:
Am I right that 4 properties enable `skipEqualsCheck`? Why would you ever want to skip the equals check?

Reply:
Also, doesn't the `Light` class do an equals check? Is there some inefficiency there?

Reply:
I am not sure this is all old code I think i will do a second pass to clean it up - This PR was just mainly to remove the type fixing for LightComponent but I can extend it to a LightComponent refactor

Reply:
Yeah, probably best not to expand the remit of this PR beyond the bare minimum otherwise it could introduce bugs.